code/datums/materials/_material.dm

```diff
@@ -38,6 +40,9 @@ Simple datum which is instanced once per type and is used for every object of sa
 	if(material_flags & MATERIAL_ADD_PREFIX)
 		source.name = "[name] [source.name]"
 
+	if(beauty_modifier)
+		addtimer(CALLBACK(source, /datum.proc/AddComponent, /datum/component/beauty, beauty_modifier * amount), 0)
```

Comment:
Why exactly does this need to be async?

Reply:
honest to god I just got this from the OG implementation from back in the day and I remember me getting stuck on this for ages and being told by ninja or something that i should make it async.